agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/sapnwdi/NwdiManifestLocatingTransformer.java

```diff
@@ -0,0 +1,95 @@
+package com.teamscale.jacoco.agent.commit_resolution.sapnwdi;
+
+import com.teamscale.client.CommitDescriptor;
+import com.teamscale.jacoco.agent.options.sapnwdi.DelayedSapNwdiMultiUploader;
+import com.teamscale.jacoco.agent.options.sapnwdi.SapNwdiApplications;
+import com.teamscale.jacoco.agent.util.LoggingUtils;
+import com.teamscale.report.util.ClasspathWildcardIncludeFilter;
+import org.conqat.lib.commons.string.StringUtils;
+import org.slf4j.Logger;
+
+import java.lang.instrument.ClassFileTransformer;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.security.CodeSource;
+import java.security.ProtectionDomain;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * {@link ClassFileTransformer} that doesn't change the loaded classes but searches their corresponding Jar/War/Ear/...
+ * files for a git.properties file.
+ */
+public class NwdiManifestLocatingTransformer implements ClassFileTransformer {
+
+	private final Logger logger = LoggingUtils.getLogger(this);
+	private final DelayedSapNwdiMultiUploader store;
+	private final ClasspathWildcardIncludeFilter locationIncludeFilter;
+	private final Map<String, SapNwdiApplications.SapNwdiApplication> markerClassesToApplications;
+
+	public NwdiManifestLocatingTransformer(
+			DelayedSapNwdiMultiUploader store,
+			ClasspathWildcardIncludeFilter locationIncludeFilter,
+			Collection<SapNwdiApplications.SapNwdiApplication> apps) {
+		this.store = store;
+		this.locationIncludeFilter = locationIncludeFilter;
+		this.markerClassesToApplications = apps.stream().collect(
+				Collectors.toMap(sapNwdiApplication -> sapNwdiApplication.getMarkerClass().replace('.', '/'),
+						application -> application));
+	}
+
+	@Override
+	public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass,
+							ProtectionDomain protectionDomain, byte[] classFileContent) {
+		if (className == null || !className.startsWith("eu")) {
+			return null;
+		}
+		logger.info(
+				"Found " + className + " protection avail " + (protectionDomain != null) + " " + classFileContent.length);
```

Comment:
This is hard to understand. Was that for initial testing? Then please remove. If it might be relevant for users, please rephrase the log message.
In case you remove it, please make sure to add the class name to the log message in line 61. Otherwise that does not provide enough information to be useful.

Reply:
Yes as well. Totally forgot about the log messages up here.